chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/state/JoinedRoomsReceivedIntegrationTest.kt

```diff
@@ -0,0 +1,234 @@
+package com.pusher.chatkit.rooms.state
+
+import com.pusher.chatkit.rooms.api.JoinedRoomApiType
+import com.pusher.chatkit.rooms.api.JoinedRoomApiTypeMapper
+import com.pusher.chatkit.rooms.api.RoomMembershipApiType
+import com.pusher.chatkit.rooms.api.RoomReadStateApiType
+import com.pusher.chatkit.state.LeftRoom
+import com.pusher.chatkit.state.ReconnectJoinedRoom
+import com.pusher.chatkit.state.RoomUpdated
+import com.pusher.chatkit.state.State
+import com.pusher.chatkit.users.api.UserApiType
+import com.pusher.chatkit.users.api.UserSubscriptionDispatcher
+import com.pusher.chatkit.users.api.UserSubscriptionEvent
+import com.pusher.chatkit.util.DateApiTypeMapper
+import io.mockk.confirmVerified
+import io.mockk.mockk
+import io.mockk.verify
+import org.reduxkotlin.Dispatcher
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+internal object JoinedRoomsReceivedIntegrationTest : Spek({
```

Comment:
minor: for tests there's no need for visibility modifiers are tests are independent and not exposed anywhere